src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -54,24 +54,53 @@ public async Task<ImmutableArray<CodeAction>> AddConstructorParametersFromMember
             }
         }
 
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates if:
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        private IMethodSymbol GetDelegatedConstructorBasedOnParameterNames(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
```

Comment:
why do these members need to be here, instead of just being helper methods in the 'State' class.  Also, can they be static as well?

Reply:
Seems like they should belong to the 'state' class but I think they might stay non-static as a private method？ Since they are only used during the initialization